### DISCUSSION TITLE
Notifications in case of loosing connection

### DISCUSSION THREAD

**Discussion by Beatle95** [microsoft/msquic#3129] [Q&A] [answered]
Hello. I am using MsQuic with QUIC_PARAM_CONN_SEND_BUFFERING set to FALSE. I am not storing pointers for allocated send buffers anywhere, just pass them to StreamSend and free in send_complete callback. If there is a lot of pending data to be sent on the stream, but connection lost, will I get notifications for all buffers, that will not be sent anymore to free them?

**Comment by nibanks** [comment, marked as answer]
Yes, you will always get a cancelled notification if the stream/connection is shut down prematurely.
